Parquet/src/main/java/io/deephaven/parquet/ParquetFileReader.java

```diff
@@ -75,6 +77,56 @@ public ParquetFileReader(String filePath, SeekableChannelsProvider channelsProvi
         type = fromParquetSchema(fileMetaData.schema,fileMetaData.column_orders);
     }
 
+    private HashSet<String> columnsWithDictionaryUsedOnEveryDataPage = null;
+    public HashSet<String> getColumnsWithDictionaryUsedOnEveryDataPage() {
+        if (columnsWithDictionaryUsedOnEveryDataPage == null) {
+            columnsWithDictionaryUsedOnEveryDataPage = calculateColumnsWithDictionaryUsedOnEveryDataPage();
+        }
+        return columnsWithDictionaryUsedOnEveryDataPage;
+    }
+
+    private HashSet<String> calculateColumnsWithDictionaryUsedOnEveryDataPage() {
+        final HashSet<String> result = new HashSet<>(fileMetaData.getSchemaSize());
+        int rowGroupSeq = 0;
+        for (RowGroup rowGroup : fileMetaData.getRow_groups()) {
```

Comment:
What kind of monster wrote this library? Camel case mixed with underscores? ;-)

Reply:
Is code generation, for thrift.   Click on the method or type and take a peek at the file you land on.  Don't do it on a full stomach.